samples/python/kinetics/custom_reactions.py

```diff
@@ -25,41 +25,67 @@
 custom_reactions = [r for r in reactions]
 custom_reactions[2] = ct.Reaction(
     equation='H2 + O <=> H + OH',
-    rate=lambda T: 38.7 * T**2.7 * exp(-3150.15428/T))
+    rate=lambda T: 38.7 * T**2.7 * exp(-3150.1542797022735/T))
+custom_reactions[4] = ct.Reaction(
+    equation='H2O2 + O <=> HO2 + OH',
+    rate=lambda T: 9630.0 * T**2.0 * exp(-2012.8781339950629/T))
 
 gas1 = ct.Solution(thermo='ideal-gas', kinetics='gas',
                    species=species, reactions=custom_reactions)
 
 # construct reactions based on ExtensibleRate: replace 2nd reaction with equivalent
 # ExtensibleRate
-@ct.extension(name="extensible-Arrhenius")
+class ExtensibleArrheniusData(ct.ExtensibleRateData):
+    __slots__ = ("T",)
+    def __init__(self):
+        self.T = None
+
+    def update(self, gas):
+        T = gas.T
+        if self.T != T:
+            self.T = T
+            return True
+        else:
+            return False
```

Comment:
While this may go beyond the scope of this PR (it is based on choices made in the C++ core), it is likely confusing to users to return a boolean. For C++, things are well hidden so the added complexity needed for avoiding unnecessary evaluations doesn't matter. For `ExtensibleRateData`, it may be easier to ignore things, meaning:
```Python
def update(self, gas):
    self.T = gas.T
```